src-ui.v3/src/PkgId.hs

```diff
@@ -144,4 +147,16 @@ verToText (Ver x) = T.pack . Ver.showVersion . Ver.makeVersion $ x
 
 ----------------------------------------------------------------------------
 newtype TagN = TagN { tagNToText :: Text }
-  deriving (Eq,Ord,FromJSON,ToJSON,ToJSONKey,FromJSONKey,FromHttpApiData,ToHttpApiData)
\ No newline at end of file
+  deriving (Eq,Ord,FromJSON,ToJSON,ToJSONKey,FromJSONKey,FromHttpApiData,ToHttpApiData)
+
+data Matches = Matches 
+  { matchesInput :: Text
+  , matchesExact :: Map.Map Text ()
+  , matchesInfix :: Map.Map Text (Text,Text,Text)
+  }
+  deriving (Eq,Ord)
+
+matchesEmpty :: Matches
+matchesEmpty = Matches { matchesInput = T.empty :: T.Text, matchesExact = Map.empty, matchesInfix = Map.empty} 
```

Comment:
why the redundant `:: T.Text` type-sig?

Reply:
i'll remove it. At first it is `""` then I found the `T.empty` but forgot to remove it.